### ISSUE TITLE
Enter the Gungeon 64 bits crashes on strcmp

### ISSUE THREAD

**Issue by DevilBlackDeath** [ptitSeb/box64#703]
Hi,
So one game I fully expected running well was Enter The Gungeon as it was made to run on box86 according to the compatibility list. Unfortunately it crashes at execution. I've run other Unity games such as Tricky Towers without problem so it seems weird.

Tried running it on box86 but I've got no luck so far running 32 bit mono apps on aarch64 with box86. Unfortunately I think it comes down to being unable to install the armhf equivalent of mono-complete or at the very least mono-runtime or mono-devel leading to usually "corrupt stacktrace" or similar issues. 

Here is the box64 log of execution with rolling log. Tell me if you need any other log.
[etg_box64.txt](https://github.com/ptitSeb/box64/files/11241805/etg_box64.txt)
(crashes with a segmentation fault at the end)

Should I open a box86 issue for the inability to run mono apps on aarch64 or is it a completely unavoidable problem ?

**Comment by ptitSeb** [comment]
I'm versy confused on what is not working here...
is EtG x64 not working with box64
or EtG x86 not working with box86

EtG is a Unity3D games, and will not try to use any native libmono or mono componant.

I tried quickly on my side, and `EtG.x86_64` start fine with box64, but `EtG.x86` doesn't start with box86, and stay stuck at a black screen.

**Comment by DevilBlackDeath** [comment]
Both don't work on an aarch64 environment. Hmmm then this may mean it's the same issue as I have with wine. Might be some dependency ? As you can see from the log (running EtG x64 on box64) there doesn't seem to be missing symbols (aside from a few X11 ones, but EtG x86 have the same missing ones and at least goes to the black screen).

Forgot to specify it's all again running on my Pi 4 4GB with latest Raspbian x54 up to date.
Could it be related to maybe some of the most basic libraries like libc (since the crash seems to indicate it was caused by strcmp in libc, and the crash happening when running wine64 is pretty similar) ?

**Comment by ptitSeb** [comment]
Did you launch with `MESA_GL_VERSION_OVERRIDE=3.2 ./EtF.x86_64` or something similar? I tried on a Pythium ARM mini PC running Debian on myside. I haven't started my PI/400 for now, so I don't know if it works there, but Iaunched other Unity3D game in the past, and wine stuff too, without intermitant issues.
Do you overclock your PI?

**Comment by DevilBlackDeath** [comment]
Yes I did, based on your recommendation in the box86 compatibility list, I expected that to be a Unity thing. That was also needed for running Tricky Towers actually. It wouldn't start without it either. I'll try to see if I have another x64 Unity-based game in my library with which to test as Tricky Towers is unfortunately x86 only on Linux.

As far as overclocking goes I've tested both with and without. I've had to reset my distro for those tests multiple times so I often come back in clean, upgraded with only the required dependency for wine and box respectively (most of which, if not all, are installed as aarch64/armhf multiarch). I end up overclocking only when some game or app needs it.

**Comment by DevilBlackDeath** [comment]
Here is the fullest log I can think of in box64 (rolling log, log level 2 and dlsym errors) :
[etg_box64_full.txt](https://github.com/ptitSeb/box64/files/11242356/etg_box64_full.txt)
Only "weird" thing I can see is "my_dlsym_error" being called but it's called three times and only crashes after the third so I'd be surprised it's caused by this (especially since its return code seems to be 0, which is usually the "error code" for "no error")

**Comment by LukeShortCloud** [comment]
@DevilBlackDeath is this still an issue with the latest version of Box64?
